### DISCUSSION TITLE
Swoole service not running

### DISCUSSION THREAD

**Discussion by navjotjsingh** [shlinkio/discussions#75] [Help wanted] [answered]
#### How Shlink is set-up

* Shlink Version: 2.9.3
* PHP Version: 8.0.12
* Distro: Ubuntu 20.04.3
* How do you serve Shlink: Self-hosted swoole
* Database engine used: MariaDB (10.6.5)

#### Summary

I followed all the steps from the guide and installed Swoole by following the guide over at https://www.swoole.co.uk/docs/get-started/installation

I then used your guide to install and enable the service.

But I get the following error when I try to enable the service.

```
sudo update-rc.d shlink_swoole enable
Failed to enable unit: Unit /run/systemd/generator.late/shlink_swoole.service is transient or generated.
```

And on starting the service, the process never runs. I can't connect to the server.

```
curl localhost:8080 -i
curl: (7) Failed to connect to localhost port 8080: Connection refused
```

```
ps aux | grep shlink_swoole
navjot     83087  0.0  0.0   9028   608 pts/0    S+   09:44   0:00 grep --color=auto shlink_swoole
```

**Comment by acelaya** [comment, marked as answer]
Not sure if this could help https://serverfault.com/questions/1054662/systemd-failing-enable-with-service-is-transient-or-generated-whats-wrong-wi

That error deeply depends on how your server is setup.

**Comment by cdn0x12** [comment]
try `cp /run/systemd/generator.late/shlink_swoole.service /etc/systemd/system/shlink_swoole.service`
